src/users/users.service.ts

```
@@ -1,26 +1,109 @@
-import { Injectable } from '@nestjs/common';
-import { CreateUserDto } from './dto/create-user.dto';
-import { UpdateUserDto } from './dto/update-user.dto';
-
+import { HttpException, HttpStatus, Injectable } from '@nestjs/common';
+import { Prisma } from '@prisma/client';
+import { PrismaService } from 'src/prisma/prisma.service';
+import * as bcrypt from 'bcrypt'
 @Injectable()
 export class UsersService {
-  create(createUserDto: CreateUserDto) {
-    return 'This action adds a new user';
+  constructor(private prisma: PrismaService) {}
+  async createOne(createUserDto: Prisma.UserCreateInput) {
+    try {
+      const res = await this.prisma.user.findUnique({
+        where: {email: createUserDto.email}
+      })
+      if (res) {
+        throw new HttpException('user already exists', HttpStatus.BAD_REQUEST)
+      }
+      const hashpassowrd = await bcrypt.hash(createUserDto.password, 10)
+      createUserDto.password = hashpassowrd
+      const user = await this.prisma.user.create({data: createUserDto})
+      return user
+    } catch (error) {
+      console.log(error);
+      throw new HttpException(error, HttpStatus.BAD_REQUEST);
+    }
+  }
+
+  async findAll() {
+    try {
+      const users = await this.prisma.user.findMany();
+      return users;
+    } catch (error) {
+      throw new HttpException(`Error : ${error}`, HttpStatus.BAD_REQUEST);
+    }
   }
 
-  findAll() {
-    return `This action returns all users`;
+  async findOne(id: number) {
+    try {
+      const user = await this.prisma.user.findUnique({
+        where: { id: id },
+      });
+      return user;
```

user can be null here, please throw not found error if that happened.